### ISSUE TITLE
Fill Missing Animations From JaidynReiman's Assets

### ISSUE THREAD

**Issue by jrconway3** [sanderfrenken/Universal-LPC-Spritesheet-Character-Generator#140]
With the new animations in place now:
https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/issues/88

There's several sets of resources I have released that have these animations, but they need to be mapped to the new expanded ULPC assets:
https://opengameart.org/content/lpc-1-hairstyle-2-hair-extensions-3-previously-unofficially-released-hairstyles
https://opengameart.org/content/lpc-new-elven-ear-variants
https://opengameart.org/content/lpc-update-now-animated-relm-hairstyle-pieces-xlong-ponytail-bangs-and-braids-headband
https://opengameart.org/content/lpc-furry-ears-tails-for-rpg-sprites
https://opengameart.org/content/lpc-new-wing-animations-for-elizawys-batfeathered-wings

All of these animations are done for Idle/Sit/Run/Jump. Climb is not done but should be fairly easy.

**Comment by jrconway3** [comment]
Some updates:

I mapped my XLong Ponytail and Tied Back Hairstyles to all frames. I started working on adding animations to my older hairstyles and redesigning them slightly to look a bit nicer, but that's taking longer than I anticipated so I'll get back to that later.


I have filled out the older ears (long, big, elven) to ALL frames, INCLUDING adding them to child finally! In addition, I mapped all my ears to all frames on child, still working on adult. I think I'm just going to open a PR solely for the ears and I'll open PR's piece-by-piece, because its taking a long time to map even the easiest assets to all frames. Plus the PR's inevitably wind up extremely massive and hard to manage.

**Comment by jrconway3** [comment]
Ears are all completely done now. I think I'll resume hairstyles again tomorrow.

**Comment by jrconway3** [comment]
I'd like to open a PR without it relying on waiting for the hairstyles to be merged first, but I'd have to recreate all the changes I made to the html structure and I'm almost out of free time for today. So I probably won't be able to do it yet. If it just gets merged anyway then its irrelevant. That said Sander Franken might be waiting for bluecarrot to chime in.

**Comment by jrconway3** [comment]
I DID actually open that PR, actually:
https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/pull/151

So this is just the ear fixes and nothing else.

**Comment by jrconway3** [comment]
There's a few problems with some of the ears. I noticed the "dragon" ears have some bad positioning on, for example, some of the new combat animations. I'll have to test and fix these issues as I go.

**Comment by jrconway3** [comment]
I fixed that ear positioning now. What I'm still missing at this point in time are all the head band/hair tie pieces (I think I'll split off the red portion of these into a separate layer that can be overlayed with any other color now), the tails and wings, and the remaining hair animations. These hair animations are rather annoying so I'll take these a bit slower. I really do want the ponytails to move during the run animation.

The Hair Extensions should be easy to do, as these are already animated anyway except for the short braid, which I think I only really need to animate much on run and jump.

**Comment by jrconway3** [comment]
Working on tails now. I originally wanted to do hair next but I'll do tails now instead because I already started.

**Comment by jrconway3** [comment]
In addition to my own tails, though, I'll also be adding in Nilla's Lizard tails and adding extra recolors. As well as adding these tails to child as well.


Tails aren't TOO hard, because I already mapped the animations to run, jump, and sit. I have enough different tail frames to do the combat animations, and "fluffy wolf" is already done. I'll simply be porting these tails to the other frames, and child doesn't have anything other than the frames I already mapped anyway.


I'm considering introducing a custom animation where the tails wag back and forth for 8 frames (mine only). I added this alternate animation but did not implement it yet. The default idle is only 2 frames and I will be mapping normal idles here which I didn't do before.

**Comment by jrconway3** [comment]
All of my tails are done now, I just want to finish Nilla's lizard tails before opening a PR.

**Comment by jrconway3** [comment]
Wings are now mapped to all frames. Though I think I forgot to add myself to the credits for the Lizard Wings, its not a huge deal; I didn't create any new frames for Lizard Wings anyway. And I already added my name to the Feathered/Bat wings when I added these before.

**Comment by jrconway3** [comment]
I need to create a list of things to finish. I know there's Capes, the old Hairstyles, and the Kimono. I think maybe the Hair Extensions aren't done yet, either, so I should implement those, too.

The Kimono is obviously the hardest one.

**Comment by jrconway3** [comment]
Oof, apparently the original entry I have not only includes the cape, but also the tunic/sleeveless shirt. I have new sleeveless shirts but I'd like to clean up these other ones eventually, too, and I really liked that tunic style. Clothing in general is a PITA. I'll probably not prioritize that right away and just do the rest of the cards.

I'm thinking I'll do a quick paste-over of all the past hairstyles I did without animations just to get it out of the way. If I do animations, I'll make them very simplistic. I'm not that fond of those older hairstyles anyway. I'd like to redo some of them as well (I started reshading them), but I never finished that. However, I think I'd prefer to keep the old ones as well as a separate option just because they've been around so long.

**Comment by jrconway3** [comment]
Started working on adding hair extensions to all frames. Apparently I never did this before. I cleaned up the simple bang a little bit. The only thing that's left here is the combat animations, though I still need to add in the braids as well. I'd kinda like to get all the hairstyles out in one quick go.

**Comment by jrconway3** [comment]
XLong Bangs are on adult, all three sets of existing hair extensions are updated to all child frames, and I'll work on the braids next.

I thought about renaming Hair Extensions to "braids" as that's basically what they are, they're bangs of different styles but stylized in certain ways. However, the naming convention would mess with existing hairstyles, so I dropped that idea.


I was going to try and include my new unfinished hairstyles as well... for for now, I'm just going to drop all the hair extensions I have in one PR before moving onto something else after they're done. I really want to spend a lot of time figuring out some good animations for extra long hairstyles so they don't look so flat.

**Comment by jrconway3** [comment]
Just about done with the basic braids. These ones didn't have any animations on jump/run. I finished jump/run now. I just need to do combat on the basic braids, then finish the xlong braids and we should be good.

For the xlong braids, only combat wasn't completed... but I never properly tested jump/run. I want to fully test them and clean them up before saying they're "done".


I might have time to work on new hairstyles tomorrow. These aren't related to this issue, they're new hairstyles I designed single poses for but never finished:
https://ko-fi.com/jaidynreiman?viewimage=II2I5XGI39#galleryItemView

Two wavy extra long hairstyles (one based on xlong ponytail, but without the ponytail), and a new wavy/curly style, along with new sets of hair extensions to go with them.

**Comment by jrconway3** [comment]
I ended up working on new hairstyles last night. Well, one new hairstyle. XLong Wavy 2 is almost done, I took inspiration from Nilla122's "XLong" in order to align the blowing hair animation for the cast a bit cleaner and nicer than my original. I still need to do run, and run is going to be a bit complicated. I somewhat know how I want the "waving" animation to go. It won't be the same as XLong Ponytail.

XLong Wavy (which is based on XLong Ponytail, but with the hair down), shouldn't be too hard. I just need to borrow elements for the run animation from the original XLong Ponytail. In that case, though, I will try and reflect the original wave, it just needs to be meshed to the whole head instead of just a ponytail. In addition, I probably will redo cast as well.

Wavy Curly I have no idea how to handle just yet. I think I want to make a shorter version of this, so this may have two versions. Either way the animation will be a PITA, so I might hold off on this one for some time and instead work on other hairstyles.


For example, Nilla122's XLong. I looked at the animations and am currently figuring out how best to clean it up and map to all frames. I think like with my hairstyles, I want to unify male/female and add a "background" layer.

**Comment by jrconway3** [comment]
Going to open a PR for the hair extensions tonight for sure. I just finished the long braids, the short braids will be relatively quick to finish the combat animations (its just a simple back and forth wave; the long braids were harder because they had to mesh around the arms).

That said, I am working on other things now, too. I am probably going to be increasing the amount of FEGBA stuff I do, which may make things a bit slower here.

I would like to close this card ASAP, but before I move onto the other hairstyles in this card, I'm going to finish three XLong Hairstyles:
Nilla122's XLong
XLong Wavy
XLong Wavy 2

And new sets of Hair Extensions:
XLong Wavy Bangs (both sides)
XLong Wavy Bangs 2 (both sides)
Straight Bangs (both sides)

I'd like to have these done within a week or so. That mostly depends on how distracted I get with FEGBA spriting.

**Comment by jrconway3** [comment]
Lol I haven't even gone back to doing my hairstyle variants yet... Its so stupid, too. I only need to create animations where it waves back and forth. But I'm nervous about how to handle that "wave" animation effect... I know I want it to be different from the original XLong Ponytail (XLong Wavy I'll based its wave on XLong Ponytail).

The basic idea behind the wave should be a fairly slight sway back and forth. I need to figure out a good way to handle it. One idea I had is having the darker and lighter colors gradually move down as the hair sways, but that I tried and it looks really hard to pull off. Which is part of the reason why I haven't gotten back to it yet.


Then there's the other hairstyles... I'm not too worried about them, but I do want to go back and finish them. The Ponytails need to have a movement animation for Run, Jump, and Combat. Maybe a SLIGHT move animation for walk, too.

Side Swoop needs to have the swoops sway a bit, and for Jump/Run they should go up and down a decent amount.

**Comment by jrconway3** [comment]
I think I want to close this card right away. I'm going to ignore my very first release, which wasn't even my release at all, and just get this released updated to all frames:
https://opengameart.org/content/lpc-1-hairstyle-2-hair-extensions-3-previously-unofficially-released-hairstyles


For the first release, I'll just do those separately. So once I map the above to all frames, I'll close this card.


The only things left after those hairstyles are my tunics and tube tops, as well as capes. Those assets I'd rather handle separately. For the tunics and tube tops, I want to gradually complete all the assets leading up to the tunics and tube tops they are derived from. For the capes, I want to create brand new capes as well (a larger, wider cape in addition to the regular capes). Those are separate projects, so I'll worry about those later.


Once I finish the Plate body armor, I'll work on finishing this up next, where I just finish the hairstyles on all frames and then move on, closing this card.

**Comment by jrconway3** [comment]
Lmao forgot about the Kimono. That will take a lot more work. I can work on those assets slowly though. The hair isn't too bad.

My plan atm is to complete both my original hairstyles on all frames as well as the redrawn versions. The redrawn versions I'll drop on OGA as a separate release when I introduce brand new hairstyles though.

**Comment by jrconway3** [comment]
Note to self:
Create reduced color Chainmail Armour and Hood. I'd like the ability to create recolors of these, but it is very complicated with the current color scheme.

I noticed checking through the history that Bluecarrot12 already did reduced color variations of the plate armour and other assets. Chainmail looks very complicated to reduce color but I think we can do it. That said, I should probably focus on mapping it to all frames first before recoloring... however, that will be a massive PITA to do so with this massive palette. So it might not be all that viable to do so.

**Comment by jrconway3** [comment]
I'm heavily considering making the ponytails be a Hair Extension instead. Put them over any hairstyle you want. I think it could work... I would need to add extra assets to the w/e/s views to allow the ponytail to extend off of the main hairstyle. I'll experiment around with how it works and see if its possible to make this work in that way.


I'm unsure at this time how I want to handle the current hairstyles. I redrew the older hairstyles, but I don't want to replace the old ones completely. So I figure I just have to do both sets separately. At the same time, some parts don't look THAT different, so maybe its not worth it. I already dropped the Short Topknot asset and am only using the new one; this topknot only had an extra lighter shade on it, so it wasn't worth keeping the old version.


I have two versions of the north hair (without the ponytail). One version (the new one) has slightly extra lightest shading on the bottom half of the hair. The side and south views, however, have basically been completely redrawn at this time. I much prefer the new ones, but I'm keeping the old ones as well.


Jewfro was already done, but I tried to add some subtle movements with a few skews to the hair on run.

**Comment by jrconway3** [comment]
Yeah doing skews for run works well for some hairstyles. I can probably do something similar for Curly Long as well, I really want to get that one out, too.

**Comment by jrconway3** [comment]
Okay I'm almost done with the two Topknots. I am hesitant at this time to push out my redesigned XLong Topknot, so I think I'd rather hold off on that a bit longer.

Right now, Jewfro and both "base" Topknot styles are done. I have renamed the base styles to Side Parted w/Bangs. The revamped version is Side Parted w/Bangs 2, I'm keeping both because the redesign is significantly different.

The ponytails themselves, however, are not THAT different. They are slightly more detailed than the originals, but I didn't feel the need to keep the old ones. So I'm just going to replace the old ponytails and make them a separate asset, a Hair Extension that can be placed over any other hairstyle.

They are designed for Side Parted w/Bangs, of course, but they even work pretty well with both Tied Back (which I may rename as well, maybe its more of a Bob variation?) and even side swoop. They won't necessarily work with all hairstyles, so perhaps I should keep a version of them that has the ponytail still built-in.



Right now, Short Topknot is finished on everything except Run at this point. Long Topknot I'm currently working on Run right now as well, but it also still needs Combat, too.


I haven't started animating Side Swoop at all yet but it is mapped to all frames. I'm mostly just going to add run and jump as per usual. Maybe I could add some movement to walk and slash, but I'm not sure at this time.


I'm hoping to also add all these hairstyles to child, too.

**Comment by jrconway3** [comment]
PR is open for these hairstyle updates. Now that I've added a separate asset for ponytails I will probably do the same thing with the XLong Ponytail and add it to the Hair Extensions > Ponytails category, but I'll do that later.

I did a few renamings but everything looks correct to me...


The only thing left for this card is the Kimono assets, but it will likely take some time to finish it. I'll work on that on/off with the armour updates.

**Comment by jrconway3** [comment]
Obi is done and Tabi Socks are almost done. I plan to also add these to Child, too, though. However, the Obi Knot is not yet done... I want to have a nice flowing animation for run so that may take some extra effort.

I was considering quickly going through the rest of the Belts really quick as well, but IDK how long that will take. I really do want to do it before doing the Leather Armour because I plan to also extract that Belt, assuming the existing Leather Belt isn't the same one. If it is the same one, then doing the Belt first will make it easier to finish the Leather Armour.

Yup, its not the same one. So here's what I'll do: I'll finish my assets (Obi, Obi Knot, Tabi Socks, and the Boots), then I'll do the rest of the belts as the next update and include the Leather Armour Belt as well. Then I'll start doing Leather Armour.

**Comment by jrconway3** [comment]
I got distracted on something else lol. But Tabi Socks are done, I just have to drop the Obi Knot in and I'll do a release. I want to do a "flipped" version of the Obi Knot as well so it can be included on either side. I'm not dropping in the Boots yet, I'll do them later.

**Comment by jrconway3** [comment]
I skipped Child this time because I decided I'll do Child assets only after I add some more new bases.

**Comment by jrconway3** [comment]
This stuff is mostly done anyway. The remaining issues I have created separate cards for here:
https://github.com/jrconway3/Universal-LPC-Spritesheet-Character-Generator/issues/8
https://github.com/jrconway3/Universal-LPC-Spritesheet-Character-Generator/issues/9
